functions/Install-DbaSqlWatch.ps1

```diff
@@ -0,0 +1,231 @@
+            if (Get-Command -ErrorAction SilentlyContinue -Name "Expand-Archive") {
+                try {
+                    Expand-Archive -Path $LocallyCachedZip -DestinationPath $LocalCacheFolder -Force
+                } catch {
+                    Stop-Function -Message "Unable to extract $LocallyCachedZip. Archive may not be valid." -ErrorRecord $_
+                    return
+                }
+            } else {
+                # Keep it backwards compatible
+                $shell = New-Object -ComObject Shell.Application
+                $zipPackage = $shell.NameSpace($LocallyCachedZip)
+                $destinationFolder = $shell.NameSpace($LocalCacheFolder)
+                Get-ChildItem "$LocalCacheFolder\SqlWatch.zip" | Remove-Item
+                $destinationFolder.CopyHere($zipPackage.Items())
+            }
+
+            Write-Message -Level VeryVerbose "Deleting $LocallyCachedZip"
+            Remove-Item -Path $LocallyCachedZip
+        }
+
+    }
+
+
+    process {
+        if (Test-FunctionInterrupt) {
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            if ($PSCmdlet.ShouldProcess($instance, "Installing SqlWatch")) {
+                try {
+                    Write-Message -Level VeryVerbose -Message "Connecting to $instance."
+                    $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
+                } catch {
+                    Stop-Function -Message "Failure." -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+                }
+
+                Write-Message -Level Verbose -Message "Starting installing/updating SqlWatch in $database on $instance."
+
+                try {
+
+                    # create a publish profile and publish DACPAC
+                    $DacPacPath = Get-ChildItem -Filter "SqlWatch.dacpac" -Path $LocalCacheFolder -Recurse | Select-Object -ExpandProperty FullName
+                    $PublishOptions = @{
+                        RegisterDataTierApplication = $true
+                    }
+                    $DacProfile = New-DbaDacProfile -SqlInstance $server -Database $Database -Path $LocalCacheFolder -PublishOptions $PublishOptions | Select-Object -ExpandProperty FileName
+                    $PublishResults = Publish-DbaDacPackage -SqlInstance $server -Database $Database -Path $DacPacPath -PublishXml $DacProfile
+
+                    # parse results
+                    $parens = Select-String -InputObject $PublishResults.Result -Pattern "\(([^\)]+)\)" -AllMatches
+                    if ($parens.matches) {
+                        $ExtractedResult = $parens.matches | Select-Object -Last 1 #| ForEach-Object { $_.value -replace '(', '' -replace ')', '' }
+                    }
+                    [PSCustomObject]@{
+                        ComputerName         = $PublishResults.ComputerName
```

Comment:
before 1.0, i'd like to align our `Install-` commands so that they all have similar output. Can you please update this one to something along the lines of

````powershell
$baseres = @{
                            ComputerName = $server.ComputerName
                            InstanceName = $server.ServiceName
                            SqlInstance  = $server.DomainInstanceName
                            Database     = $Database
                            Name         = $script.BaseName
                        }
                        if ($scriptError) {
                            $baseres['Status'] = 'Error'
                        } elseif ($script.BaseName -in $allprocedures) {
                            $baseres['Status'] = 'Updated'
                        } else {
                            $baseres['Status'] = 'Installed'
                        }
                        [PSCustomObject]$baseres
````

Reply:
Output object updated.